### PR TITLE
Support for tagging and cross account access

### PR DESCRIPTION
**Pull request by excessivedemon**
* Added support for tagging (filtering volumes to backup based on their tags) via the --tag parameter

Examples of usage:

- to backup volumes with a tag of "Name" and value of "foo": `backup-monkey --region us-west-2 --tag Name:foo`

- to backup volumes with a tag of "Name" and value of "foo", and tag of "environment" and value of "production": `backup-monkey --region us-west-2 --tag Name:foo environment:production`

- to backup volumes with a tag of "Name" and value of "foo" or "bar": `backup-monkey --region us-west-2 --tag Name:"['foo','bar']"`

* Added support for --reverse-tags (Use the --tags parameter as a blacklist instead of a whitelist)

Example of usage:

- to snapshot all instances with a tag `Name` and value is NOT `foo`: `backup-monkey --region us-west-2 --tag Name:foo --reverse-tags`

* Added support for cross account snapshots. Added the following parameters:

`--cross-account-number CROSS_ACCOUNT_NUMBER`

Do a cross-account snapshot. Requires that you pass in the --cross-account-role parameter

`--cross-account-role CROSS_ACCOUNT_ROLE`

The role backup-monkey will assume when doing a cross-account snapshot

* Setup RPM builds to require python, python-setuptools, and RPM install to require python-boto. Setup post-install to install pip.

* Updated codes to run on Python 2.6 (Boto supports Python 2.6)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -36,8 +39,19 @@ Usage
...
+                            do not have a `Name` tag with the value `foo`
+      --reverse-tags        Do a reverse match on the passed in tags. E.g. --tag
+                            Name:foo --reverse-tags will snapshot all instances
+                            that do not have a `Name` tag with the value `foo`
+      --cross-account-number CROSS_ACCOUNT_NUMBER
+                            Do a cross-account snapshot. Requires that you pass in
+                            the --cross-account-role parameter
+      --cross-account-role CROSS_ACCOUNT_ROLE
+                            The role backup-monkey will assume when doing a cross-
+                            account snapshot
```
Comment (pas256) [inline README.rst]: Can you please add an example here. It is not clear if you want the name of the role or the ARN.

File: setup.py [hunk 2]
```diff
@@ -63,8 +68,13 @@
         'Intended Audience :: System Administrators',
         'License :: OSI Approved :: Apache Software License',
         'Natural Language :: English',
-        'Programming Language :: Python :: 2.7',
```
Comment (pas256) [inline setup.py]: Keep both of these here, 2.7 and 2.6.
Thanks.

**Comment by pas256** [comment]
Thanks @excessivedemon. Let me know once those changes are taken care of and I'll merge.

Are you going to be at re:invent?

**Comment by excessivedemon** [comment]
Thanks @pas256 I have now added examples for the --cross-account-number and --cross-account-role parameters, and updated the README.rst and setup.cfg files. There's also a small bug fix I added for the cross account functionality.

I'm currently in New Zealand but will be watching via live stream :smile:

**Comment by pas256** [comment]
Wicked. Thanks

**Pull request merged**